compiler/cli/cli-base/src/org/jetbrains/kotlin/cli/jvm/modules/CliJavaModuleFinder.kt

```diff
@@ -119,7 +120,13 @@ class CliJavaModuleFinder(
     private fun findSystemModule(moduleRoot: VirtualFile, useSig: Boolean = false): JavaModule.Explicit? {
         val file = moduleRoot.findChild(if (useSig) PsiJavaModule.MODULE_INFO_CLASS + ".sig" else PsiJavaModule.MODULE_INFO_CLS_FILE)
             ?: return null
-        val moduleInfo = JavaModuleInfo.read(file, javaFileManager, allScope) ?: return null
+        val moduleInfo = try {
+            JavaModuleInfo.read(file, javaFileManager, allScope) ?: return null
+        } catch (e: IllegalStateException) {
+            reportError(e.message!!)
+            throw CompilationErrorException()
```

Comment:
This way we'll lose all important information from `e.cause` that would help us in diagnosing the problem. How about we at least append something like `"\nCaused by: ${e.cause::class.java.name}: ${e.cause.message}"` to the error?

Reply:
Agree, updated.